tapdb/universe_stats_test.go

```diff
@@ -115,21 +115,29 @@ func TestUniverseStatsEvents(t *testing.T) {
 
 	ctx := context.Background()
 
-	const numAssets = 3
+	const numTranches = 3
 
-	sh := newUniStatsHarness(t, numAssets, db.BaseDB, statsDB)
+	sh := newUniStatsHarness(t, numTranches, db.BaseDB, statsDB)
+
+	// Record the number of units in this asset.
+	var numUnits uint64
+	for i := 0; i < numTranches; i++ {
+		numUnits += sh.universeLeaves[i].Leaf.Amt
+	}
 
 	// Before we insert anything into the DB, we should have all zeroes for
-	// the main set of stats.
+	// the main events.
 	sh.assertUniverseStatsEqual(t, universe.AggregateStats{
-		NumTotalAssets: 0,
-		NumTotalProofs: 0,
-		NumTotalSyncs:  0,
+		NumTotalAssetUnits:    numUnits,
+		NumTotalAssetTranches: numTranches,
+		NumTotalGroups:        numTranches,
```

Comment:
Confused on how `NumTotalGroups == numTranches` is always true here. in `newUniStatsHarness` we randomly use a group key for the universe ID, so shouldn't `NumTotalGroups` also be 0 half the time?

Reply:
Ah yes, I needed to add `WHERE group_key IS NOT NULL`, should now be correct.